### PR TITLE
git: use linux-headers@5.15

### PR DESCRIPTION
**Pull request by danielnachun**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/git.rb [hunk 1]
```diff
@@ -28,7 +29,7 @@ class Git < Formula
   uses_from_macos "zlib", since: :high_sierra
 
   on_linux do
-    depends_on "linux-headers@4.4"
+    depends_on "linux-headers@5.15"
```
Comment (carlocab) [inline Formula/git.rb:32]: Why is this needed? And, in particular, why are the headers still needed at runtime?

Comment (danielnachun) [reply Formula/git.rb:32]: The runtime dependency is weird - it could just be an oversight.  I will do a local test to see why it’s needed.

File: Formula/git.rb [hunk 2]
```diff
@@ -4,6 +4,7 @@ class Git < Formula
   url "https://mirrors.edge.kernel.org/pub/software/scm/git/git-2.37.2.tar.xz"
   sha256 "1c3d9c821c4538e7a6dac30a4af8bd8dcfe4f651f95474c526b52f83406db003"
   license "GPL-2.0-only"
+  revision 1
```
Comment (carlocab) [inline Formula/git.rb:7]: If it's a build dependency, do we need the `revision` bump?

Comment (danielnachun) [reply Formula/git.rb:7]: We don't need one anymore - I forgot to remove this.

**Review by carlocab — APPROVED** [review]
